informer/spec.go

```diff
@@ -0,0 +1,19 @@
+package informer
+
+import (
+	"k8s.io/client-go/pkg/runtime"
+)
+
+// Observer functions are called when the cache.ListWatch List and Watch
+// functions are called.
+type Observer interface {
+	OnList()
+	OnWatch()
+}
+
+// ZeroObjectFuncs provides zero values of an object and objects' list ready to
+// be decoded. The provided zero values must not be reused by zeroObjectFactory.
+type ZeroObjectFactory interface {
+	NewObject() runtime.Object
+	NewObjectList() runtime.Object
+}
```

Comment:
I know. But it's immediately obvious that you need `ZeroObjectFactory` when you use `New`. I spend time looking for `ZeroObjectFactory` definition, which I expected to be in `zero_object.go`. If you had `InformerFactory` interface that would make sense. My 2c. I stop this discussion now. Do whatever you think.